### PR TITLE
Documentation changes for 0.3.0

### PR DESCRIPTION
**Pull request by prashastia**
* Modified Readme
* Added BigQueryTableExample
* Successful [Cloudbuild](https://pantheon.corp.google.com/cloud-build/builds;region=global/f0cca6af-10ed-49f2-a774-8e6eb0bfa2c9;step=12?inv=1&invt=AbXrnQ&project=testproject-398714)
[Edit]
Merge PR #144 before reviewing and merging this PR.

/gcbrun

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 The connector supports streaming data from [Google BigQuery](https://cloud.google.com/bigquery/) tables to Apache Flink, 
 and writing results back to BigQuery tables.
-This is done by using the [Flink’s Datastream API](https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/dev/datastream/overview/) 
+This is done by using the [Flink’s Datastream API](https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/dev/datastream/overview/) as well as [Flink's Table API and SQL](https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/dev/table/overview/)
```
Comment (jayehwhyehentee) [inline README.md:8]: This is supported via Datastream and Table API ...

File: README.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ granted.
 * Unix-like environment (we use Linux, Mac OS X)
 * Git
 * Maven (we recommend version 3.8.6)
-* Java 8
+* Java 11
```
Comment (jayehwhyehentee) [inline README.md:88]: Undo

File: README.md [hunk 3]
```diff
@@ -96,9 +96,9 @@ There are two ways to access the connector.
 The connector is available on the [Maven Central](https://repo1.maven.org/maven2/com/google/cloud/flink/)
 repository.
 
-| Flink version | Connector Artifact                                                       |
-|---------------|--------------------------------------------------------------------------|
-| Flink 1.17.x  | `com.google.cloud.flink:flink-1.17-connector-bigquery:0.2.0`             |
+| Flink version | Connector Artifact                                                                                                          |
+|---------------|-----------------------------------------------------------------------------------------------------------------------------|
+| Flink 1.17.x  | `com.google.cloud.flink:flink-1.17-connector-bigquery:0.2.0`,  `com.google.cloud.flink:flink-1.17-connector-bigquery:0.3.0` |
```
Comment (prashastia) [inline README.md:101]: Add another row.

Comment (jayehwhyehentee) [inline README.md:101]: Fix indentation. Separate rows

File: README.md [hunk 4]
```diff
@@ -150,10 +151,11 @@ Follow [this document](https://cloud.google.com/dataproc/docs/concepts/component
 | 0.1.0-preview                  | ✓   | ✓   |
 | 0.2.0-preview                  | ✓   | ✓   |
 | 0.2.0                          | ✓   | ✓   |
+| 0.3.0                          | ✓   | ✓   |
 
 ## Usage
-
-The connector uses Flink’s Datastream API, and can be used in Java applications. For a Flink source, it offers two read 
+The connector could be used via Flink’s Datastream API, and can be used in Java applications. 
+It could also be used via Flink's Table API and SQL. For a Flink source, it offers two read
```
Comment (jayehwhyehentee) [inline README.md:158]: The connector can be used with Flink's Datastream and Table APIs in Java applications. The source offers two read modes, bounded and unbounded. The sink offers at-least-once delivery gaurantee.

File: README.md [hunk 5]
```diff
@@ -175,7 +177,8 @@ modes, bounded and unbounded.
 * Source factory methods are defined at `com.google.cloud.flink.bigquery.source.BigQuerySource`.
 * Source configs are defined at `com.google.cloud.flink.bigquery.source.config.BigQueryReadOptions`.
 * BigQuery connection config is defined at `com.google.cloud.flink.bigquery.common.config.BigQueryConnectOptions`.
-* Sample Flink application using connector is defined at `com.google.cloud.flink.bigquery.examples.BigQueryExample`.
+* Sample Flink application using connector is defined at `com.google.cloud.flink.bigquery.examples.BigQueryExample` for the Datastream API
+  and at `com.google.cloud.flink.bigquery.examples.BigQueryTableExample` for the Table API and SQL.
```
Comment (prashastia) [inline README.md:181]: Add SQL example in `BigQueryTableExample`

File: README.md (hunk 2 above)
Comment (prashastia) [reply README.md]: Fixed.

File: README.md (hunk 1 above)
Comment (prashastia) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (prashastia) [reply README.md]: Fixed.

File: README.md (hunk 4 above)
Comment (prashastia) [reply README.md]: Fixed.

File: README.md [hunk 6]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+        .table(...)
+        .project(...)
+        .dataset(...)
+        .testMode(...)
+        .credentialAccessToken(...)
+        .credentialFile(...)
+        .credentialKey(...)
+        .sinkParallelism(...) // Should be atmost 128
+        .deliveryGuarantee(...)
+        .build();
```
Comment (jayehwhyehentee) [inline README.md:432]: Please mention "required" or "optional" against these parameters. Also, add a point (with link) to refer to table of sink configs for more details. Same applicable to source as well

File: README.md [hunk 7]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+```java
+// Insert entries in this sinkTable
+TableResult res = sourceTable.executeInsert("bigQuerySinkTable");
+// wait for the job to complete 
+// (for jobs running on dataproc cluster via "gcloud dataproc submit" command only) 
+res.await();
+```
+#### More Details:
+* Input and Output tables (catalog tables) must be registered in the TableEnvironment.
+* Moreover, the schema of the registered table must match the schema of the query.</b>
```
Comment (jayehwhyehentee) [inline README.md:454]: Remove "Moreover"

File: README.md [hunk 8]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+* [BigQueryTableSchemaProvider](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/main/flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/BigQueryTableSchemaProvider.java) is a helper class which contains the method `getTableDescriptor()` which could be used to obtain a [TableDescriptor](https://nightlies.apache.org/flink/flink-docs-master/api/java/org/apache/flink/table/api/TableDescriptor.html) for creation of catalog table via `BigQueryTableConfig` (`BigQuerySinkTableConfig` for sink options and `BigQueryReadTableConfig` for read options). 
+Users could also create their own catalog tables; provided the schema of the registered table, and the associated BigQuery table is the same.
+* Limitations:
+    * Inability to read and then write `TIME` type BigQuery records. Reading `TIME` type records and subsequently writing them to BigQuery would result in an error due to misconfigured types between BigQuery and Flink's RowData.
```
Comment (jayehwhyehentee) [inline README.md:462]: Elaborate that this happens only if source is BigQuery

File: README.md [hunk 9]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+    * Incorrect value obtained during read and write of `BIGNUMERIC` type BigQuery Records. Reading `BIGNUMERIC` type records from a BigQuery table and subsequently writing them to BigQuery would result in incorrect value being written to BigQuery as Flink's RowData does not support NUMERIC Types with precision more than 38 (BIGNUMERIC supports precision upto 76).
```
Comment (jayehwhyehentee) [inline README.md:463]: Same as above

File: README.md [hunk 10]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+| `partitionDiscoveryRefreshIntervalInMinutes` | Integer           | Periodicity (in minutes) of partition discovery in table. This config is used in unbounded source. If unspecified, the default value used is 10 minutes.                                                                                                                                                                                                  | `BigQueryReadTableConfig`                            |
+| `sinkParallelism`                            | Integer           | Integer value indicating the parallelism for the sink . This config is used in unbounded source and is optional. If unspecified, the application decides the optimal parallelism. <br/>Maximum value: 128.                                                                                                                                                | `BigQuerySinkTableConfig`                            |
```
Comment (clmccart) [inline README.md:477]: nit: extra space after "sink" before "."

**Review by clmccart — COMMENTED** [review]
review of README. will take a look at the rest of the PR soon

File: README.md [hunk 11]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+
+#### Sink: At Least Once
+
+Flink [Sink](https://nightlies.apache.org/flink/flink-docs-release-1.17/api/java/org/apache/flink/api/connector/sink2/Sink.html)
+is the base interface for developing a sink. With checkpointing enabled, it can offer at-least-once consistency. Our
+implementation uses BigQuery Storage's [default write stream](https://cloud.google.com/bigquery/docs/write-api#default_stream)
+in Sink's [Writers](https://nightlies.apache.org/flink/flink-docs-release-1.17/api/java/org/apache/flink/api/connector/sink2/SinkWriter.html).
+
+```java
+// StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
```
[the remark this answers is not in the document]
Comment (prashastia) [reply README.md:320]: Yes, this is a part of setup, users can choose to declare BatchEnvironment or any others based on their needs.

File: README.md [hunk 12]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+// final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+// env.enableCheckpointing(CHECKPOINT_INTERVAL);
+// final StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+// Create the Config.
+BigQueryTableConfig readTableConfig =  new BigQueryReadTableConfig.Builder()
+        .table(...) // REQUIRED
+        .project(...) // REQUIRED
+        .dataset(...) // REQUIRED
+        .partitionDiscoveryInterval(...) // OPTIONAL; only in CONTINUOUS_UNBOUNDED source
```
[the remark this answers is not in the document]
Comment (prashastia) [reply README.md:393]: Provided in the table below, is that fine ?

File: README.md [hunk 13]
```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
...
+| `dataset`                                    | String            | Dataset containing the table. This config is required for standard tables, but not when loading query results.                                                                                                                                                                                                                                            | `BigQueryReadTableConfig`, `BigQuerySinkTableConfig` |
+| `table`                                      | String            | BigQuery table in the format `[[projectId:]dataset.]table`. This config is required for standard tables, but not when loading query results.                                                                                                                                                                                                              | `BigQueryReadTableConfig`, `BigQuerySinkTableConfig` |
```
[the remark this answers is not in the document]
Comment (prashastia) [reply README.md:467]: Fixed.

File: README.md (hunk 10 above)
Comment (prashastia) [reply README.md:477]: Fixed.

**Review by clmccart — APPROVED** [review]

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (clmccart) [reply README.md:393]: yeah that's probably sufficient; thanks!

**Review by jayehwhyehentee — APPROVED** [review]

**Pull request merged**